odl/discr/lp_discr.py

```diff
@@ -1248,6 +1267,275 @@ def discr_sequence_space(shape, exponent=2.0, impl='numpy', **kwargs):
+    Giving one of ``min_corner`` and ``max_corner`` together with
+    ``cell_sides`` keeps ``nsamples`` constant:
+
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                max_corner=[3, 3])
+    uniform_discr([1.0, 1.0], [3.0, 3.0], [10, 5])
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                max_corner=[3, 3]).cell_sides
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                nsamples=[5, 5])
+    uniform_discr([1.0, 1.0], [1.5, 3.0], [5, 5])
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                cell_sides=[1, 1])
+    uniform_discr([1.0, 1.0], [11.0, 6.0], [10, 5])
+
+    Specifying 3 or more new parameters will ignore ``discr``:
+
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                max_corner=[3, 3],
+                                       cell_sides=[0.5, 0.25])
+    uniform_discr([1.0, 1.0], [3.0, 3.0], [4, 8])
+
+    Finally, everything can be done per axis. ``None`` entries are
+    interpreted as "not specified":
+
+    >>> uniform_discr_fromdiscr(discr, min_corner=[None, 1],
+    ...                                max_corner=[3, None],
+                                       cell_sides=[None, 0.25])
+    uniform_discr([2.0, 1.0], [3.0, 2.25], [10, 5])
+    """
+    if not isinstance(discr, DiscreteLp):
+        raise TypeError('`discr` {!r} is not a DiscreteLp instance'
+                        ''.format(discr))
+    if not discr.partition.is_regular:
+        raise ValueError('`discr` {} is not uniformly discretized'
+                         ''.format(discr))
+
+    # Normalize partition parameters
+    min_corner = normalized_scalar_param_list(min_corner, discr.ndim,
+                                              param_conv=float, keep_none=True)
+    max_corner = normalized_scalar_param_list(max_corner, discr.ndim,
+                                              param_conv=float, keep_none=True)
+    nsamples = normalized_scalar_param_list(nsamples, discr.ndim,
+                                            param_conv=safe_int_conv,
+                                            keep_none=True)
+    cell_sides = normalized_scalar_param_list(cell_sides, discr.ndim,
+                                              param_conv=float, keep_none=True)
+
+    nodes_on_bdry = kwargs.pop('nodes_on_bdry', False)
+    nodes_on_bdry = normalized_nodes_on_bdry(nodes_on_bdry, discr.ndim)
+
+    new_beg = []
+    new_end = []
+    new_shape = []
+    new_csides = []
+    for i, (b, e, n, s, old_b, old_e, old_n, old_s) in enumerate(
+            zip(min_corner, max_corner, nsamples, cell_sides,
+                discr.min_corner, discr.max_corner, discr.shape,
+                discr.cell_sides)):
+        num_params = sum(p is not None for p in (b, e, n, s))
+
+        if num_params == 0:
```

Comment:
I'll check that. However, I explicitly flattened it at some point because I got confused with the cases.